javaparser-core/src/main/java/com/github/javaparser/TokenRange.java

```diff
@@ -0,0 +1,38 @@
+package com.github.javaparser;
+
+import static com.github.javaparser.utils.Utils.assertNotNull;
+
+/**
+ * The range of tokens covered by this node.
```

Comment:
That feels like it has its own set of tokens, where it really is just two pointers indicating a range in the full list of tokens.

Actually, I would say that adding Range was a mistake and it should have been this implementation from the start - so this should become the new Range :)